### DISCUSSION TITLE
SOC-based charging and discharging control

### DISCUSSION THREAD

**Discussion by studiowena** [Sleeper85/esphome-jk-bms-can#50] [Ideas]
Hi,

As we know, it is best to use batteries in the flat range of 80-20%. It would be great if we could set when REBULK loading should resume based on SOC %. Likewise, we can set it to charge up to 80% SOC. V-base alignment is imprecise

**Comment by Sleeper85** [comment]
Hi,

The SoC of bms is not more precise than the voltage. It could be an idea if the soc calculation comes from a shunt.

**Reply by studiowena** [reply]
If, for example, we have a discharge cutoff at V and a stronger momentary load comes, the battery will be disconnected even though there is still, for example, 40% SOC. That's why I'm wondering how to control it conveniently. Of course, we have HA and automation, but it would be better if the Inverter did it

**Reply by MrPabloUK** [reply]
The CAN Automatic Discharge Current switch will handle this situation.
As the minimum cell voltage approaches cut-off value (as set in the BMS), maximum discharge current is reduced.

This should prevent momentary high load from causing sufficient voltage sag to trip 0% SOC and battery disconnect.
